astropy/coordinates/tests/test_iau_fullstack.py

```diff
@@ -110,72 +75,53 @@ def test_iau_fullstack(
+    # where altitude is high enough.  The SOFA guides imply that below 5 deg is
     # where accuracy gets more problematic, and testing reveals that alt<~0
-    # gives garbage round-tripping, and <10 can give ~1 arcsec uncertainty
-    if fullstack_obsconditions[0].value == 0:
-        # but if there is no refraction correction, check everything
-        msk = slice(None)
-        tol = 5 * u.microarcsecond
-    else:
-        msk = aacoo.alt > 5 * u.deg
-        # most of them aren't this bad, but some of those at low alt are offset
-        # this much.  For alt > 10, this is always better than 100 masec
-        tol = 750 * u.milliarcsecond
+    # gives garbage round-tripping, and <10 can give ~1 arcsec uncertainty,
+    # but if there is no refraction correction, we still check everything
+    msk = aacoo.alt > min_alt
 
     # now make sure the full stack round-tripping works
     icrs2 = aacoo.transform_to(ICRS())
 
-    adras = np.abs(fullstack_icrs.ra - icrs2.ra)[msk]
-    addecs = np.abs(fullstack_icrs.dec - icrs2.dec)[msk]
-    assert np.all(
-        adras < tol
-    ), f"largest RA change is {np.max(adras.arcsec * 1000)} mas, > {tol}"
-    assert np.all(
-        addecs < tol
-    ), f"largest Dec change is {np.max(addecs.arcsec * 1000)} mas, > {tol}"
+    assert_quantity_allclose(
+        np.abs(FULLSTACK_ICRS.ra - icrs2.ra)[msk], 0 * u.μas, atol=tol, rtol=0
+    )
+    assert_quantity_allclose(
+        np.abs(FULLSTACK_ICRS.dec - icrs2.dec)[msk], 0 * u.μas, atol=tol, rtol=0
+    )
 
     # check that we're consistent with the ERFA alt/az result
-    iers_tab = iers.earth_orientation_table.get()
-    xp, yp = u.Quantity(iers_tab.pm_xy(fullstack_times)).to_value(u.radian)
-    lon = fullstack_locations.geodetic[0].to_value(u.radian)
-    lat = fullstack_locations.geodetic[1].to_value(u.radian)
-    height = fullstack_locations.geodetic[2].to_value(u.m)
-    jd1, jd2 = get_jd12(fullstack_times, "utc")
-    pressure = fullstack_obsconditions[0].to_value(u.hPa)
-    temperature = fullstack_obsconditions[1].to_value(u.deg_C)
-    # Relative humidity can be a quantity or a number.
-    relative_humidity = u.Quantity(fullstack_obsconditions[2], u.one).value
-    obswl = fullstack_obsconditions[3].to_value(u.micron)
-    astrom, eo = erfa.apco13(
-        jd1,
-        jd2,
+    astrom, _ = erfa.apco13(
+        *get_jd12(fullstack_times, "utc"),
         fullstack_times.delta_ut1_utc,
-        lon,
-        lat,
-        height,
-        xp,
-        yp,
-        pressure,
-        temperature,
-        relative_humidity,
-        obswl,
+        (geodetic := fullstack_locations.geodetic).lon.rad,
+        geodetic.lat.rad,
+        geodetic.height.to_value(u.m),
+        *Angle(iers.earth_orientation_table.get().pm_xy(fullstack_times)).rad,
+        altazframe.pressure.to_value(u.hPa),
+        altazframe.temperature.to_value(u.deg_C),
+        altazframe.relative_humidity.to_value(u.one),
+        altazframe.obswl.to_value(u.μm),
```

Comment:
It does look nice!

Reply:
It is worth pointing out that pressure, temperature and obswl could be looked up from `obsconditions` too, but `obsconditions["relative_humidity"].to_value(u.one)` does not work if `fullstack_obsconditions` uses a `float`, whereas the `altazframe` attribute is always a `Quantity`.